prov/verbs/src/fi_verbs.h

```diff
@@ -283,6 +310,16 @@ typedef int(*fi_ibv_mr_dereg_cb)(struct fi_ibv_mem_desc *md);
 
 struct fi_ibv_mem_notifier;
 
+struct fi_ibv_xrc_domain {
+	int				xrcd_fd;
+	struct ibv_xrcd			*xrcd;
+
+	/* The domain maintains a RBTree for mapping an endpoint destination
+	 * addresses to physical XRC INI QP connected to that host. */
+	fastlock_t			ini_mgmt_lock;
+	struct ofi_rbmap		*ini_conn_rbmap;
```

Comment:
Yes perhaps, but it would introduce the interaction of two locks. One coarse grain lock for creating/deleting/searching RB Nodes, then one for handling pending connection processing within each node. The latter is important, until a physical connection is created, subsequent connections to the same node must be queued. If the first physical connection is rejected by the RxM peer, then the subsequent connection must create the physical connection.